subprojects/gradle/instrumentation-tests/build.gradle.kts

```diff
@@ -36,7 +36,7 @@ dependencies {
     implementation(project(":gradle:kotlin-dsl-support"))
     implementation(project(":gradle:process"))
     implementation(project(":gradle:runner:client"))
-    implementation(project(":gradle:runner:shared-test"))
```

Comment:
This dependency changes inputs in compile tasks and breaks caching.

Couldn't find the exact reasons. I've suspected `random` functions but couldn't prove it. Only removing test dependencies helped.
Anyway, we don't want to have test dependencies in build scripts.